### ISSUE TITLE
Corrupted dwarf debug data when include_sources=true

### ISSUE THREAD

**Issue by AndrewSB** [getsentry/symbolic#749]

**Comment by kamilogorek** [comment]
cc @Swatinem @loewenheim

**Comment by Swatinem** [comment]
This is interesting. `include-sources` basically tries to iterate through all the files included in the debug data.
Even if you do not use `include-sources`, the upload itself might succeed, but we will hit that same error on our backend when trying to use the file.

Can you share the file with us? We would have to take a closer look at it to figure out whats going on here.
Though `Hit the end of input before it was expected` is an indication that maybe one of the headers has a wrong value. Either because the file is indeed broken, or because our parsing does not handle it correctly.

**Comment by AndrewSB** [comment]
yeah! which file exactly? is there a command i can run that will list the files you're interested in that i can then zip & email over to one of you?

**Comment by Swatinem** [comment]
```
[17:16:53]: > Found 262 debug information files
```

Oh well, without some more context in the error message, it might indeed be hard to figure out which of these 262 files it is choking with. For the time being, can you try to focus the command to specific sub-directories that contain less individual files? I would say bisecting this manually is the way to go for now.

**Comment by AndrewSB** [comment]
<img width="691" alt="Screenshot 2023-02-17 at 3 56 04 PM" src="https://user-images.githubusercontent.com/3814772/219818379-393c99e5-d15f-47cf-b43f-411a26cca304.png">

how do i constrain the files it's looking at? this is my invocation (minus the auth token). i think it's using a bunch of context variables from fastlane, so i don't think i can just run in the terminal, but i'd love to do that instead if i can

**Comment by Swatinem** [comment]
`sentry-cli` has recently added some debug logging to help you find the file that caused this error.

**Comment by ashwoods** [comment]
Hi @AndrewSB, do you need further assistance with this issue?

**Comment by ashwoods** [comment]
Closing due to inactivity.
